HyperVExtension/src/HyperVExtension/Providers/HyperVProvider.cs

```diff
@@ -93,6 +84,25 @@ public ComputeSystemAdaptiveCardResult CreateAdaptiveCardSessionForComputeSystem
         return new ComputeSystemAdaptiveCardResult(notImplementedException, OperationErrorString, notImplementedException.Message);
     }
 
-    // This will be implemented in a future release, but will be available for Dev Environments 1.0.
-    public ICreateComputeSystemOperation CreateCreateComputeSystemOperation(IDeveloperId developerId, string inputJson) => throw new NotImplementedException();
+    /// <summary> Creates an operation that will create a new Hyper-V virtual machine. </summary>
+    public ICreateComputeSystemOperation? CreateCreateComputeSystemOperation(IDeveloperId? developerId, string inputJson)
+    {
+        try
+        {
+            if (JsonSerializer.Deserialize(inputJson, typeof(VMGalleryCreationUserInput)) is not VMGalleryCreationUserInput userInput)

```

Comment:
When is this possible? If json contains "null" only?
The only value of non-VMGalleryCreationUserInput type returned by JsonSerializer.Deserialize would be null (which can be treated as VMGalleryCreationUserInput ).
To me, it's easier to read this if it would be something like 
var userInput = JsonSerializer.Deserialize(inputJson, typeof(VMGalleryCreationUserInput)) ?? throw...
instead of using "is not VMGalleryCreationUserInput" which implies that there could be returned some type other than VMGalleryCreationUserInput.

Reply:
for your first statement "When is this possible? If json contains "null" only?". Technically anyone can send the provider a string. So, the Serializer not being able to deserialize the json into an object we expect could possibly make the result null. Looking at the example here: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/deserialization, it seems that the .Net folks expect that deserialization could possibly end up returning a null value.

when you say: 
"
instead of using "is not VMGalleryCreationUserInput" which implies that there could be returned some type other than VMGalleryCreationUserInput.
"

I guess in the future my thinking here would be to update this method to look like below, when we add more creation flows other than using the VM Gallery (e.g a more manual way providing iso files) .
E.g

```C#
if (JsonSerializer.Deserialize(inputJson, typeof(VMGalleryCreationUserInput)) is VMGalleryCreationUserInput galleryInput)
{
    // return operation to create VM from gallery
}
if (JsonSerializer.Deserialize(inputJson, typeof(VMManualCreationUserInput)) is VMManualCreationUserInput manualInput)
{
   // return operation to create VM manually with user provided input
}

// throw
```

but it costs nothing for me to make it more readable for now. I can update it.

Reply:
updated thanks